internal/github/publish_github_gist.go

```diff
@@ -0,0 +1,202 @@
+package github
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"log/slog"
+
+	gh "github.com/google/go-github/v58/github"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+	"go.opentelemetry.io/otel/attribute"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+	"github.com/blackstork-io/fabric/print"
+	"github.com/blackstork-io/fabric/print/htmlprint"
+	"github.com/blackstork-io/fabric/print/mdprint"
+)
+
+func makeGithubGistPublisher(loader ClientLoaderFn) *plugin.Publisher {
+	return &plugin.Publisher{
+		Doc:  "Publishes content to github gist",
+		Tags: []string{},
+		Config: dataspec.ObjectSpec{
+			&dataspec.AttrSpec{
+				Name:        "github_token",
+				Type:        cty.String,
+				Constraints: constraint.RequiredNonNull,
+				Secret:      true,
+			},
+		},
+		Args: dataspec.ObjectSpec{
```

Comment:
Perhaps you can add some constraints/default values? This helps with our documentation generation (not yet done for publishers, but would be in future) and allows you to skip repetetive calls like checks for Nulls in your code

Reply:
@Andrew-Morozko As mentioned in the issue here https://github.com/blackstork-io/fabric/issues/134, these arguments are optional.

Reply:
https://github.com/blackstork-io/fabric/pull/222/commits/5ad895c3109be03645f373635f3fcbbdfd21eda6